File: JavaScript/Binary Search/374. Guess Number Higher or Lower.js
/* 374. Guess Number Higher or Lower
Difficulty: Me
Submission Link: https://leetcode.com/problems/guess-number-higher-or-lower/submissions/1376857960/ */

var guessNumber = function (n) {
    let right = n
    let left = 0
    while (left <= right) {
        let mid = Math.floor((right + left) / 2)

        const result = guess(mid)
        if (result > 0) {
            left = mid + 1
        } else if (result < 0) {
            right = mid
        } else {
            return mid
        }
    }

    return -1
};

File: C#/Binary Search/374. Guess Number Higher or Lower.cs
/* 374. Guess Number Higher or Lower
Difficulty: Me
Submission Link: https://leetcode.com/problems/guess-number-higher-or-lower/submissions/1376858798/ */

public class Solution : GuessGame {
    public int GuessNumber(int n) {
        int low = -1;
        int high = n;

        if(n == 2147483647) {
            return n;
        }

        while(low <= high) {
            int mid = low + (high - low) / 2;
            int result = guess(mid);

            if(result == 0) {
                return mid;
            } else if (result == -1) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }
        

        return -1;
    }
}